package/kedro_viz/models/flowchart.py

```diff
@@ -586,52 +587,17 @@ def set_viz_metadata(cls, _, values):
 
         return None
 
-    # TODO: improve this scheme.
-    def is_plot_node(self):
-        """Check if the current node is a plot node.
-        Currently it only recognises one underlying dataset as a plot node.
-        In the future, we might want to make this generic.
-        """
-        return self.dataset_type in (
-            "plotly.plotly_dataset.PlotlyDataset",
-            "plotly.json_dataset.JSONDataset",
-            "plotly.plotly_dataset.PlotlyDataSet",
-            "plotly.json_dataset.JSONDataSet",
-        )
-
-    def is_image_node(self):
-        """Check if the current node is a matplotlib image node."""
-        return self.dataset_type == "matplotlib.matplotlib_writer.MatplotlibWriter"
-
-    def is_metric_node(self):
-        """Check if the current node is a metrics node."""
-        return self.dataset_type in (
-            "tracking.metrics_dataset.MetricsDataset",
-            "tracking.metrics_dataset.MetricsDataSet",
-        )
-
-    def is_json_node(self):
-        """Check if the current node is a JSONDataset node."""
-        return self.dataset_type in (
-            "tracking.json_dataset.JSONDataset",
-            "tracking.json_dataset.JSONDataSet",
-        )
-
-    def is_tracking_node(self):
-        """Checks if the current node is a tracking data node"""
-        return self.is_json_node() or self.is_metric_node()
-
-    def is_preview_node(self):
-        """Checks if the current node has a preview"""
-        if not (self.viz_metadata and self.viz_metadata.get("preview_args", None)):
-            return False
-
-        return True
-
     def get_preview_args(self):
         """Gets the preview arguments for a dataset"""
         return self.viz_metadata.get("preview_args", None)
 
+    def is_preview_disabled(self):
+        """Checks if the dataset has a preview disabled"""
+        if self.viz_metadata:
+            preview_value = self.viz_metadata.get("preview")
+            return preview_value is not None and not preview_value
+        return False
```

Comment:
```suggestion
    def is_preview_disabled(self):
        """Checks if the dataset has a preview disabled"""
        if self.viz_metadata:
            preview_value = self.viz_metadata.get("preview")
            return preview_value is not None and not preview_value
        return False
```
I found this complicated, can we assume it's True unless specified as False?

Reply:
+1

Reply:
Maybe something like this:

```suggestion
    def is_preview_disabled(self):
        """Checks if the dataset has a preview disabled"""
        return self.viz_metadata and self.viz_metadata.get("preview") is False
```

Reply:
so, the preview feature is always enabled unless the user specifies `viz_metadata: preview: false` ? @rashidakanchwala

Reply:
yup, that's true. by default

Reply:
if self.viz_metadata is None, the function will return None. Can we have a return value of boolean in all cases ? Can we do something like - `return viz_metadata is not None and viz_metadata.get("preview") is False` . Thank you

Reply:
ok have more clarity on what you and @merelcht meant. It returns None 
so i have changed it to return bool(self.viz_metadata and self.viz_metadata.get("preview") is False) -- now it will always return False except when preview: False